### ISSUE TITLE
less harsh error reporting for invalid function names

### ISSUE THREAD

**Issue by bucaran** [fish-shell/fish-shell#2565]
```fish
# either
function ""; end
# or
function $null; end
```

will throw, instead of reporting an error.

```
fish: function function_add called with null value for argument ! data.name.empty().
fish: This is a bug. Break on bugreport to debug.If you can reproduce it, please send a bug report to fish-users@lists.sf.net.
fish: Backtrace:
0   fish                                0x0000000108ff8a60 _Z15show_stackframev + 112
1   fish                                0x0000000108f8fab6 _Z15define_functionR8parser_tR12io_streams_tRKNSt3__16vectorINS3_12basic_stringIwNS3_11char_traitsIwEENS3_9allocatorIwEEEENS8_ISA_EEEERKSA_iPSA_ + 5366
2   fish                                0x00000001090183c5 _ZN25parse_execution_context_t22run_function_statementERK12parse_node_tS2_ + 501
3   fish                                0x0000000109017b94 _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t + 532
4   fish                                0x00000001090180a9 _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t + 137
5   fish                                0x000000010901b2de _ZN25parse_execution_context_t19eval_node_at_offsetEjPK7block_tRK10io_chain_t + 350
6   fish                                0x0000000108fca72c _ZN8parser_t15eval_block_nodeEjRK10io_chain_t12block_type_t + 332
7   fish                                0x0000000108fc9e55 _ZN8parser_t4evalERKNSt3__112basic_stringIwNS0_11char_traitsIwEENS0_9allocatorIwEEEERK10io_chain_t12block_type_t + 501
8   fish                                0x0000000108fd180e _Z18reader_run_commandR8parser_tRKNSt3__112basic_stringIwNS1_11char_traitsIwEENS1_9allocatorIwEEEE + 270
9   fish                                0x0000000108fd85e8 _Z11reader_readiRK10io_chain_t + 952
10  fish                                0x000000010902336a main + 4986
11  libdyld.dylib                       0x00007fff880ad5ad start + 1
12  ???                                 0x
```

Related: https://github.com/fish-shell/fish-shell/issues/2519

**Comment by zanchey** [comment]
This is a regression from 2.2.0.

**Comment by ridiculousfish** [comment]
This was fixed in ad72421b2c738e93ed8d99823974b5a823a02e03 . Thanks for reporting this.
